core/providers/anthropic/responses.go

```
@@ -284,6 +292,87 @@ func (s *anthropicToResponsesStreamState) blockType(index int) AnthropicContentB
+	state.markReasoningPayloadSent(key)
+
+	// Re-point the item's key at the new block so the deltas that follow, and the
+	// content_block_stop at output_item.done, resolve to it instead of the closed one.
+	newIdx := state.allocBlockIndex(key)
+
+	// Same signature seeding as the thinking branch at output_item.added: when the id
+	// is being smuggled through the signature, the reopened block carries it too, so a
+	// client replaying this turn folds the thinking half back together with the
+	// redacted half (see reasoningIndexByID in the ingress converter).
+	signature := ""
+	if providerUtils.ShouldEmbedReasoningItemID(ctx, bifrostResp.ExtraFields.Provider, bifrostResp.ExtraFields.RoutingInfo.Model) {
```

golangci-lint (2.12.2)

[error] SA1019: bifrostResp.ExtraFields.Provider is deprecated: use RoutingInfo.Provider. Still populated for backward compatibility; new consumers should read from RoutingInfo. 

(staticcheck)